### ISSUE TITLE
How to access storage with all my files? (.: Permission denied)

### ISSUE THREAD

**Issue by bew** [t184256/nix-on-droid#105]
Hello,

I'd like to know if there's a way to access my files from within Nix-on-Droid. 
I've tried to cd to /sdcard or /storage/emulated/0 but I can't do anything in them, ls gives `ls: .: Permission denied` for both of them. 

I've tried from the normal & failsafe app.

**Comment by Gerschtli** [comment]
Is this maybe what you need? https://github.com/t184256/nix-on-droid/issues/85

**Comment by t184256** [comment]
Yeah, you should try the storage toggle in the app settings inside Android settings.

**Comment by bew** [comment]
Thanks @Gerschtli I saw that issue but couldn't find that toggle. It was actually under "Permission".

Indeed that solves the problem, sorry for the noise I should have tried better.

Would it be possible to show a popup when trying to read these directories, or add a command to unlock them (that would show a popup for asking permission) ?

**Comment by t184256** [comment]
> Would it be possible to show a popup when trying to read these directories

Everything's possible in theory, but that'd be really hard to implement and would likely incur a performance cost.

> or add a command to unlock them (that would show a popup for asking permission)?

Yes, if someone knowing Android programming understands and revives/recreates what Termux did to achieve that.

**Comment by austinbutler** [comment]
Linking the Termux relevant wiki page: https://wiki.termux.com/wiki/Internal_and_external_storage
